onboard/import.go

```diff
@@ -37,45 +36,37 @@ type Config struct {
 	Cataloger          catalog.Cataloger
 	CatalogActions     RepoActions
 	KeyPrefixes        []string
-	Rocks              bool
 	EntryCatalog       *rocks.EntryCatalog
 }
 
 type Stats struct {
 	AddedOrChanged       int
-	Deleted              int
 	DryRun               bool
 	PreviousInventoryURL string
 	CommitRef            string
 	PreviousImportDate   time.Time
 }
 
 var (
-	ErrNoInventoryURL           = errors.New("no inventory_url in commit Metadata")
-	ErrInventoryAlreadyImported = errors.New("given inventory was already imported")
-	ErrIncompatiblePrefixes     = errors.New("prefix filter should cover at least the same keys as the previous import")
+	ErrIncompatiblePrefixes = errors.New("prefix filter should cover at least the same keys as the previous import")
 )
 
 func CreateImporter(ctx context.Context, logger logging.Logger, config *Config) (importer *Importer, err error) {
 	res := &Importer{
-		repository:         config.Repository,
 		inventoryGenerator: config.InventoryGenerator,
 		logger:             logger,
-		rocks:              config.Rocks,
 		CatalogActions:     config.CatalogActions,
 	}
 
 	if res.CatalogActions == nil {
-		res.CatalogActions = buildRepoActions(config, logger)
+		res.CatalogActions = NewCatalogRepoActions(config.EntryCatalog, graveler.RepositoryID(config.Repository), config.CommitUsername, logger, config.KeyPrefixes)
 	}
 
-	previousCommit, err := res.CatalogActions.GetPreviousCommit(ctx)
-	if err != nil {
+	if err := res.CatalogActions.InitBranch(ctx); err != nil {
 		return nil, fmt.Errorf("failed to get previous commit: %w", err)
 	}
-	res.previousCommit = previousCommit
-	shouldSort := res.previousCommit != nil || res.rocks
-	res.inventory, err = config.InventoryGenerator.GenerateInventory(ctx, logger, config.InventoryURL, shouldSort, config.KeyPrefixes)
+
+	res.inventory, err = config.InventoryGenerator.GenerateInventory(ctx, logger, config.InventoryURL, true, config.KeyPrefixes)
 	res.prefixes = config.KeyPrefixes
 	if !res.validatePrefixes() {
```

Comment:
Looks like this wouldn't work anymore (not sure if due to this PR or a previous one). Under the Importer struct, we still have a previousCommit member which is used in this method, but it is never assigned.

Reply:
This one. MVCC used that feature.

Reply:
But we still want to validate the prefixes, no?

Reply:
function name was misleading, it validated that the old and new prefixes match.
That doesn't matter anymore.